pkg/istio/servicerolebinding/servicerolebinding_test.go

```diff
@@ -40,3 +43,111 @@ func TestCreateServiceRoleBinding(t *testing.T) {
 	a.Equal("domain.one/sa/application1", serviceRoleBinding.Subjects[0].User)
 	a.Equal("domain.two/sa/application2", serviceRoleBinding.Subjects[1].User)
 }
+
+func TestParseMemberName(t *testing.T) {
+
+	cases := []struct {
+		test           string
+		member         *zms.RoleMember
+		expectedMember string
+		expectedErr    error
+	}{
+		{
+			test:           "nil member",
+			member:         nil,
+			expectedMember: "",
+			expectedErr:    fmt.Errorf("member is nil"),
+		},
+		{
+			test: "valid service member",
+			member: &zms.RoleMember{
+				MemberName: zms.MemberName("client.some-domain.dep-svcA"),
+			},
+			expectedMember: "client.some-domain/sa/dep-svcA",
+			expectedErr:    nil,
+		},
+		{
+			test: "valid user member",
+			member: &zms.RoleMember{
+				MemberName: zms.MemberName("user.somename"),
+			},
+			expectedMember: "user/sa/somename",
+			expectedErr:    nil,
+		},
+		{
+			test: "valid wildcard member",
+			member: &zms.RoleMember{
+				MemberName: zms.MemberName("user.*"),
+			},
+			expectedMember: "*",
+			expectedErr:    nil,
+		},
+	}
+
+	for _, c := range cases {
+		gotMember, gotErr := parseMemberName(c.member)
+		a := assert.New(t)
+		a.Equal(c.expectedMember, gotMember, c.test)
+		a.Equal(c.expectedErr, gotErr, c.test)
+	}
+}
+
+func TestGetServiceRoleBindingSpec(t *testing.T) {
+
+	type input struct {
+		roleName string
+		members  []*zms.RoleMember
+	}
+	cases := []struct {
+		test         string
+		input        input
+		expectedSpec *v1alpha1.ServiceRoleBinding
+		expectedErr  error
+	}{
+		{
+			test: "empty args",
+			input: input{
+				roleName: "",
+				members:  nil,
+			},
+			expectedSpec: nil,
+			expectedErr:  fmt.Errorf("no subjects found for the ServiceRoleBinding: "),
```

Comment:
shouldn't this have the role name at the end of the string?

Reply:
the passed-in role name is empty, so the error message has nothing to print